File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    // Function to get a specific row of Pascal's Triangle
    public List<Integer> getRow(int rowIndex) {
        List<Integer> ans = new ArrayList<>();
        long ansRow = 1;
        ans.add((int) ansRow); // Add the first element which is always 1
        rowIndex += 1; // Adjust for 0-based index
        
        for (int col = 1; col < rowIndex; col++) { // Loop through each column in the row
            ansRow = ansRow * (rowIndex - col); // Calculate the current element using binomial coefficient
            ansRow = ansRow / col;
            ans.add((int) ansRow); // Add the calculated element to the current row list
        }
        
        return ans; // Return the current row list
    }
}

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py

        
class Solution:
    
    def getRow(self, rowIndex: int) -> List[int]:
        ans = [1]
        def helper(n):
            res = 1
            for i in range(n):
                res *= (n - i)
                res //= i + 1
                ans.append(res)
        helper(rowIndex)
        return ans
        